csv_very_large_file.py

```diff
@@ -0,0 +1,149 @@
+@click.option('-p', '--role_mapper', type=str, required=False, default='',
+              help='If role column then role mapper in format "source_client:client,source_expert:expert,*:expert"')
+def main(filename: str):
+    file_name = filename.split("/")[-1]
+    file_name_split = file_name.split(".")[0]
+    json_file_name = file_name_split + "-hf-export.json"
+    convo_dict = combine_conversations(file_name)
+    export_to_hf_json_limit(convo_dict,json_file_name,6000)
+
+
+def combine_conversations(filename: str):
+    csv_file = filename
+    convo_dict = {}
+    no_of_unique_conversations = 0
+    no_of_utterances_total = 0
+    error_log = []
+
+    with open(csv_file, 'r', newline='', encoding='utf-8') as csvfile:
+        csvreader = csv.reader(csvfile)
+        for row_number, row in enumerate(csvreader,start=1):
+            try:
+                convo_id = row[0]
+                current_speaker = row[2]
+                current_utterance = row[3]
+            except IndexError as e:
+                error_log.append((row_number,row))
+                continue
+
+            
+            if convo_id in convo_dict:
+                #append to that text file
+                current_convo = convo_dict[convo_id]
+
+                #optional clean utterance of quotes
+                current_utterance = current_utterance.strip('"')    
+
+                #create string
+                next_str = current_speaker + ": " + current_utterance + "\n"
+
+                #append
+                current_convo = current_convo + next_str
+
+                #replace
+                convo_dict[convo_id] = current_convo
+                no_of_utterances_total += 1
+            else:
+                #adding first utterance
+                next_str = current_speaker + ": " + current_utterance + "\n"
+                convo_dict[convo_id] = next_str
+                no_of_unique_conversations += 1
+    
+    print(f"Processed Conversations: {no_of_unique_conversations}\n")
+    print(f"Processed Utterances: {no_of_utterances_total}\n")
+    print("Error utterances found: " + str(len(error_log)))
+    
+    err_file_name = "Error Log--" + csv_file
+    if(len(error_log) > 0):
+        with open(err_file_name, 'w', newline='') as err_csv_file:
+            writer = csv.writer(err_csv_file)
+            writer.writerows(error_log)
+
+    return convo_dict
+
+
+def export_to_hf_json(conversation_dict, json_file):
+    
+    examples = []
+    for conversation_id in conversation_dict:
+        example = {
+                "id": conversation_id,
+                "text": conversation_dict[conversation_id],
+            }
+        examples.append(example)
+
+    data = {
+        "$schema": "https://docs.humanfirst.ai/hf-json-schema.json",
+        "examples": examples
+    }
+
+    with open(json_file, 'w', encoding='utf-8') as jsonfile:
+        json.dump(data, jsonfile, indent=2)
+
+def export_to_hf_json_limit(conversation_dict, json_file, limit):
```

Comment:
Provide the data type of every function variables and also the data type of the values returned. If no values are returned by the function, then provide None

Add docstring to the function explaining what the function does.